backend/de.metas.business/src/main/java/de/metas/product/impexp/ProductImportHelper.java

```diff
@@ -203,10 +212,82 @@ private de.metas.adempiere.model.I_M_Product createProductRecordNoSave(@NonNull
 		product.setTrademark(importRecord.getTrademark());
 		product.setPZN(importRecord.getPZN());
 		product.setIsCommissioned(importRecord.isCommissioned());
-		product.setIsPurchased(importRecord.isPurchased());
 
 		return product;
 	}    // MProduct
 
+	public void handlePackingInstructions(@NonNull final I_I_Product importRecord, @NonNull final ProductId productId)
+	{
+		packingInstructionService.handlePackingInstructions(importRecord, productId);
+	}
+
+	public void handleCatchWeight(@NonNull final I_I_Product importRecord)
+	{
+		//	 Handle Catch Weight UOM Conversion
+		if (X_I_Product.INVOICABLEQTYBASEDON_CatchWeight.equalsIgnoreCase(importRecord.getInvoicableQtyBasedOn()))
+		{
+			handleUOMConversion(importRecord);
+		}
+	}
+
+	private void handleUOMConversion(@NonNull final I_I_Product importRecord)
+	{
+		if (importRecord.getQtyCU_UOM_ID() <=0 )
+		{
+			return;
+		}
+
+		final int fromUOM = importRecord.getQtyCU_UOM_ID();
+		final int toUOM = importRecord.getC_UOM_ID();
+		BigDecimal multiplierRate = importRecord.getUOM_MultiplierRate();
+
+		if (multiplierRate == null)
+		{
+			multiplierRate = getDefaultConversionRate(fromUOM, toUOM);
+		}
+
+		createUOMConversion(fromUOM, toUOM, multiplierRate);
+	}
+
+	private BigDecimal getDefaultConversionRate(int fromUOM, int toUOM)
+	{
+		I_C_UOM_Conversion conversion = Services.get(IQueryBL.class)
+				.createQueryBuilder(I_C_UOM_Conversion.class)
+				.addEqualsFilter(I_C_UOM_Conversion.COLUMNNAME_C_UOM_ID, fromUOM)
+				.addEqualsFilter(I_C_UOM_Conversion.COLUMNNAME_C_UOM_To_ID, toUOM)
+				.first();
+
+		return conversion != null ? conversion.getMultiplyRate() : BigDecimal.ONE;
+	}
+
+	private void createUOMConversion(int fromUOM, int toUOM, BigDecimal multiplierRate)
+	{
+		if (fromUOM <= 0 || toUOM <= 0 || multiplierRate == null || multiplierRate.signum() == 0)
+		{
+			log.warn("Skipping UOM conversion: invalid parameters (fromUOM={}, toUOM={}, multiplierRate={})",
+					fromUOM, toUOM, multiplierRate);
+			return;
+		}
+
+		I_C_UOM_Conversion conversion = queryBL
+				.createQueryBuilder(I_C_UOM_Conversion.class)
+				.addEqualsFilter(I_C_UOM_Conversion.COLUMNNAME_C_UOM_ID, fromUOM)
+				.addEqualsFilter(I_C_UOM_Conversion.COLUMNNAME_C_UOM_To_ID, toUOM)
+				.first();
+
+		if (conversion == null)
+		{
+			conversion = InterfaceWrapperHelper.newInstance(I_C_UOM_Conversion.class);
+			conversion.setC_UOM_ID(fromUOM);
+			conversion.setC_UOM_To_ID(toUOM);
+			conversion.setMultiplyRate(multiplierRate);
+			conversion.setIsActive(true);
```

Comment:
save?

Reply:
At any rate, this logic seems incorrect. Shouldn't the conversion be defined only for the imported product?

Reply:
why not using UOM Conversion DAO is used?

Reply:
yes, good point!